routemaster/models/event.rb

```diff
@@ -3,47 +3,49 @@
 require 'routemaster/mixins/assert'
 require 'base64'
 
-module Routemaster::Models
-  class Event
-    include Routemaster::Mixins::Assert
-    extend  Routemaster::Mixins::Assert
-
-    VALID_TYPES = %w(create update delete noop)
-
-    attr_reader :topic, :type, :url, :timestamp
-
-    def initialize(topic:, type:, url:, timestamp: nil)
-      _assert VALID_TYPES.include?(type), 'bad event type'
-      @topic     = topic
-      @type      = type
-      @url       = CallbackURL.new(url)
-      @timestamp = timestamp || Routemaster.now
-    end
+module Routemaster
+  module Models
+    class Event
+      include Mixins::Assert
+      extend  Mixins::Assert
+
+      VALID_TYPES = %w(create update delete noop)
+
+      attr_reader :topic, :type, :url, :timestamp
+
+      def initialize(topic:, type:, url:, timestamp: nil)
+        _assert VALID_TYPES.include?(type), 'bad event type'
```

Comment:
This is purely internal code so I'd rather err on the side of being strict. None of the app's code should use symbols here.

(I'd definitely agree in the _client_, though)